src/SignedSettlement.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity ^0.8;
+
+import {Auth} from "./Auth.sol";
+import {GPv2Trade} from "cowprotocol/libraries/GPv2Trade.sol";
+import {GPv2Interaction} from "cowprotocol/libraries/GPv2Interaction.sol";
+import {LibSignedSettlement} from "./LibSignedSettlement.sol";
+import {ECDSA} from "solady/utils/ECDSA.sol";
+import {SubPoolFactory} from "./SubPoolFactory.sol";
+import {GPv2Settlement, IERC20} from "cowprotocol/GPv2Settlement.sol";
+
+contract SignedSettlement is Auth {
+    error SignedSettlement__InvalidAttestor();
+    error SignedSettlement__DeadlineElapsed();
+
+    /// @notice whether signer can attest
+    address public immutable attestor;
+    GPv2Settlement public immutable settlement;
+
+    constructor(GPv2Settlement _settlement, address _attestor) {
+        settlement = _settlement;
+        attestor = _attestor;
+    }
+
+    /// @notice Takes the required settlement data and verifies that it has been
+    ///         signed by a `signer`, and subsequently calls `GPv2Settlement.settle`.
+    function signedSettleFullySigned(
+        address[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        GPv2Trade.Data[] calldata trades,
+        GPv2Interaction.Data[][3] calldata interactions
+    ) external {
+        (uint256 deadline, uint256 r, uint256 s, uint256 v, bytes32 digest, uint256 calldataStart, uint256 calldataSize)
```

Comment:
Why does calldatastart need to be returned from the complex computation? Don't we know it here (ie. the length of the public parameters).

Reply:
added some natspac for this